### PR TITLE
Add print barcode labels for supported types

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-1646

### CODE REVIEW

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java [hunk 1]
```diff
@@ -1,72 +1,19 @@
...
-
-import uk.ac.bbsrc.tgac.miso.core.data.Barcodable;
-import uk.ac.bbsrc.tgac.miso.core.data.Printer;
-import uk.ac.bbsrc.tgac.miso.service.PrinterService;
-
 /**
  * For when you need web utils, since this module can't access `miso-web`.
  *
  */
 public class ControllerHelperServiceUtils {
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java [hunk 2]
```diff
@@ -1,72 +1,19 @@
...
   private static final Logger log = LoggerFactory.getLogger(ControllerHelperServiceUtils.class);
```
Comment (sqwci) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java]: MAJOR Remove this unused "log" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

File: miso-web/src/main/webapp/scripts/lims.js [hunk 3]
```diff
@@ -301,6 +301,45 @@ var Utils = Utils || {
...
+                                                'Printing',
+                                                'POST',
+                                                window.location.origin + '/miso/rest/printer/' + printer.id + '?' + jQuery
+                                                    .param({
+                                                      type : type,
+                                                      ids : ids.join(',')
+                                                    })
+                                                , null, function(result) {
+//TODO: replace with Utils.showOkDialog
+alert(result == ids.length ? 'Printing successful.' : (result + ' of ' + ids.length + ' printed.'));
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/lims.js]: MINOR Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java [hunk 4]
```diff
@@ -81,10 +86,23 @@ public boolean print(byte[] content, JsonNode configuration) {
...
+  },
+  DEBUG() {
+    @Override
+    public boolean print(byte[] content, JsonNode configuration) {
+      StringBuilder buffer = new StringBuilder();
+      buffer.append("Printing debug output:");
+      for (byte b : content) {
+        buffer.append(String.format(" %02x", b));
+      }
+      log.error(buffer.toString());
```
Comment (sqwci) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java]: MAJOR Invoke method(s) only conditionally. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

**Comment by sqwci** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [box_ajax.js#L505](https://github.com/TGAC/miso-lims/blob/5f24d2ddc065de02cfc0f9534576b6a3b1bcf588/miso-web/src/main/webapp/scripts/box_ajax.js#L505): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)
1. ![MINOR][MINOR] [lims.js#L226](https://github.com/TGAC/miso-lims/blob/5f24d2ddc065de02cfc0f9534576b6a3b1bcf588/miso-web/src/main/webapp/scripts/lims.js#L226): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by callunity** [comment]
Looks good. At some point we may need to add a barcode column, but I don't think we're there yet.

To fix the failing test, update the `ListPagesIT` to account for the additional column on the List Boxes page.

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java (hunk 1 above)
Comment (djcooke) [reply spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java]: :+1:

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java (hunk 2 above)
Comment (djcooke) [reply spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java]: :+1:

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java (hunk 4 above)
Comment (djcooke) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java]: Probably should be debug-level instead

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L292](https://github.com/TGAC/miso-lims/blob/2bab26d9e0dd714ea9244ad81f08696b4c6d34cb/miso-web/src/main/webapp/scripts/lims.js#L292): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java (hunk 4 above)
Comment (apmasell) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java]: I had it that way in the beginning, but the default logging configuration makes it go nowhere. If someone sets up a debugging printer, I assume they really want to see the output.

**Comment by sqwci** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [lims.js#L292](https://github.com/TGAC/miso-lims/blob/6d320dbba3f3012f39e73cad7992c4727115bbe9/miso-web/src/main/webapp/scripts/lims.js#L292): Remove this usage of alert(...). [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
